phalcon/http/statuscode.zep

```diff
@@ -0,0 +1,182 @@
+	const CLIENT_UNORDERED_COLLECTION = 425;
+	const CLIENT_UPDATE_REQUIRED = 426;
+	const CLIENT_PRECONDITION_REQUIRED = 428;
+	const CLIENT_TOO_MANY_REQUESTS = 429;
+	const CLIENT_REQUEST_HEADER_FIELDS_TOO_LARGE = 431;
+
+	// SERVER ERROR
+	const SERVER_INTERNAL_SERVER_ERROR = 500;
+	const SERVER_NOT_IMPLEMENTED = 501;
+	const SERVER_BAD_GATEWAY = 502;
+	const SERVER_SERVICE_UNAVAILABLE = 503;
+	const SERVER_GATEWAY_TIMEOUT = 504;
+	const SERVER_HTTP_VERSION_NOT_SUPPORTED = 505;
+	const SERVER_VARIANT_ALSO_NEGOTIATES = 506;
+	const SERVER_INSUFFICIENT_STORAGE = 507;
+	const SERVER_LOOP_DETECTED = 508;
+	const SERVER_BANDWIDTH_LIMIT_EXCEEDED = 509;
+	const SERVER_NOT_EXTENDED = 510;
+	const SERVER_NETWORK_AUTHENTICATION_REQUIRED = 511;
+
+	private static _messages;
+
+	/**
+	 * Get the reason phrase for a particular status code
+	 *
+	 * @param int statusCode
+	 * @return string
+	 */
+	public static function getStatusText(int statusCode)
+	{
+		if typeof self::_messages != "array" {
+			let self::_messages = [
+				// INFORMATIONAL CODES
+				100 : "Continue",
+				101 : "Switching Protocols",
+				102 : "Processing",
+				// SUCCESS CODES
+				200 : "OK",
+				201 : "Created",
+				202 : "Accepted",
+				203 : "Non-Authoritative Information",
+				204 : "No Content",
+				205 : "Reset Content",
+				206 : "Partial Content",
+				207 : "Multi-status",
+				208 : "Already Reported",
+				// REDIRECTION CODES
+				300 : "Multiple Choices",
+				301 : "Moved Permanently",
+				302 : "Found",
+				303 : "See Other",
+				304 : "Not Modified",
+				305 : "Use Proxy",
+				306 : "Switch Proxy", // Deprecated
+				307 : "Temporary Redirect",
+				// CLIENT ERROR
+				400 : "Bad Request",
+				401 : "Unauthorized",
+				402 : "Payment Required",
+				403 : "Forbidden",
+				404 : "Not Found",
+				405 : "Method Not Allowed",
+				406 : "Not Acceptable",
+				407 : "Proxy Authentication Required",
+				408 : "Request Time-out",
+				409 : "Conflict",
+				410 : "Gone",
+				411 : "Length Required",
+				412 : "Precondition Failed",
+				413 : "Request Entity Too Large",
+				414 : "Request-URI Too Large",
+				415 : "Unsupported Media Type",
+				416 : "Requested range not satisfiable",
+				417 : "Expectation Failed",
+				418 : "I'm a teapot",
+				422 : "Unprocessable Entity",
+				423 : "Locked",
+				424 : "Failed Dependency",
+				425 : "Unordered Collection",
+				426 : "Upgrade Required",
+				428 : "Precondition Required",
+				429 : "Too Many Requests",
+				431 : "Request Header Fields Too Large",
+				// SERVER ERROR
+				500 : "Internal Server Error",
+				501 : "Not Implemented",
+				502 : "Bad Gateway",
+				503 : "Service Unavailable",
+				504 : "Gateway Time-out",
+				505 : "HTTP Version not supported",
+				506 : "Variant Also Negotiates",
+				507 : "Insufficient Storage",
+				508 : "Loop Detected",
+				511 : "Network Authentication Required"
+			];
+		}
+
+		if (array_key_exists(statusCode, self::_messages)) {
```

Comment:
isset is faster than array_key_exists also because is a language construct, not a function